pkg/integrations/v2/register.go

```diff
@@ -228,10 +228,12 @@ func MarshalYAML(v interface{}) (interface{}, error) {
 			panic(fmt.Sprintf("config not registered: %T", data))
 		}
 
-		if _, exists := uniqueSingletons[fieldName]; exists {
-			return nil, fmt.Errorf("integration %q may not be defined more than once", fieldName)
+		if integrationType == TypeSingleton {
+			if _, exists := uniqueSingletons[fieldName]; exists {
+				return nil, fmt.Errorf("integration %q may not be defined more than once", fieldName)
+			}
+			uniqueSingletons[fieldName] = struct{}{}
 		}
```

Comment:
I don't care if this changes, but I do want to point out that we don't need to introduce nested ifs here:

```suggestion
		if _, exists := uniqueSingletons[fieldName]; exists && integrationType == TypeSingleton {
				return nil, fmt.Errorf("integration %q may not be defined more than once", fieldName)
		}
		uniqueSingletons[fieldName] = struct{}{}
```

Reply:
ya I was trying to isolate the cost of resetting the value in the map but I think the legibility is worth changing it as you suggest